### ISSUE TITLE
tf.keras.optimizers.experimental.AdamW only support constant weight_decay

### ISSUE THREAD

**Issue by x10000year** [tensorflow/tensorflow#55824]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Feature Request

### Source

source

### Tensorflow Version

2.8

### Custom Code

No

### Current Behaviour?

```shell
tf.keras.optimizers.experimental.AdamW only supports constant weight decay. But usually we want the weight_decay value to decay with learning rate schedule.
```

### Standalone code to reproduce the issue

```shell
The legacy tfa.optimizers.AdamW supports callable weight_decay, which is much better.
```

### Relevant log output

_No response_</details>

**Comment by gadagashwini** [comment]
@x10000year, Thanks for proposing new feature request in TF.
Could you elaborate the feature with code snippet for better understanding. Thanks!

**Comment by x10000year** [comment]
```
model = MyModel()
schedule = tf.keras.optimizers.schedules.PolynomialDecay(
    1.0,
    decay_steps=10000)
learning_rate_fn = lambda: learning_rate * schedule(model.iterations.read_value())
weight_decay_fn = lambda: weight_decay * schedule(model.iterations.read_value())
optimizer = tfa.optimizers.AdamW(weight_decay=weight_decay_fn,
                                 learning_rate=learning_rate_fn)
```
As above code shows, tfa.optimizers.AdamW allows us to specify the schedule of weight decay, which should be in proportional with learning rate schedule. However, I cannot do the samething with tf.keras.optimizers.experimental.AdamW.

@gadagashwini

**Comment by sachinprasadhs** [comment]
Thanks for opening this issue. Development of keras moved to separate repository https://github.com/keras-team/keras/issues

Please post this issue on keras-team/keras repo.
To know more see;
https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999
Thank you!
